woocommerce/woocommerce
# Implement the components required for incentive promotion

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Partially closes #51453  .

Noting that this PR addresses the creation of the components, and will require a separate PR addressing the logic to pull and check for incentives.



### Screenshots

Link to designs: xO8YdpKfaVoP2d8NvweZ5I-fi-3106_71830

Incentive Modal:

<img width="970" alt="Screenshot 2024-11-08 at 10 41 43" src="https://github.com/user-attachments/assets/b7bfa2fb-e6c6-4078-a64e-049c74d8f540">

 
**Note** As you can see, it looks a little off what we have in the design. We have a limitation here which is that the `<Modal>` component has the close button as part of a separate "top header" section, which makes it look a little but weird. I can think of two possible solutions:

1. Remove the top header completely, and instead of the Close icon in the top right, implement a close button at the bottom next to the `Get started` button
2. Remove the top header, and implement our own custom close icon in the top right corner. I don't really like this solution, because it feels hacky, and this is something that should be addressed on the top level `Modal` component (which I don't think is in scope of the project)

cc. @orcunomattic - do you have a preference on how to address this issue?

Incentive Banner and highlighted recommendation:

<img width="2199" alt="Screenshot 2024-11-08 at 10 43 52" src="https://github.com/user-attachments/assets/eff32909-075b-4620-8c26-afbc6a4a40cb">

### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Copy the following code into the `settings-payments-main.tsx` file, replacing the existing content.

```js
/**
 * External dependencies
 */

/**
 * Internal dependencies
 */
import './settings-payments-main.scss';
import { PaymentGateways } from '~/settings-payments/components/payment-gateways';
import { OtherPaymentGateways } from '~/settings-payments/components/other-payment-gateways';
import { IncentiveBanner } from '~/settings-payments/components/incentive-banner';
import { IncentiveModal } from '~/settings-payments/components/incentive-modal';

export const SettingsPaymentsMain = () => {
	return (
		<>
			<div className="settings-payments-main__container">
				<IncentiveModal isOpen={ true } onClose={ () => {} } onSubmit={ () => {} } />
				<IncentiveBanner />
				<PaymentGateways />
				<OtherPaymentGateways />
			</div>
		</>
	);
};

export default SettingsPaymentsMain;

```

2. Run the command `pnpm run --filter=@woocommerce/plugin-woocommerce watch:build `
3. Go to the `WooCommerce > Settings > Payments` page. You should see the incentives modal. It won't close, or do anything currently.
4. Remove the `<IncentiveModal isOpen={ true } onClose={ () => {} } onSubmit={ () => {} } />` line from the code above and wait for it to recompile. 
5. Refresh the page, you should see the Incentive Banner, and the WooPayments recommendation should be highlighted.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
